ts/import-csv/MetaMapper.svelte

```diff
@@ -6,14 +6,22 @@ License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
     import * as tr from "../lib/ftl";
     import MapperRow from "./MapperRow.svelte";
 
+    let tagsIndex = 0;
+    let deckIndex = 0;
+    let notetypeIndex = 0;
+
     export let columnNames: string[];
-    export let tagsColumn: number = 0;
-    export let deckColumn: number = 0;
-    export let notetypeColumn: number = 0;
+    export let tagsColumn: number = tagsIndex - 1;
+    export let deckColumn: number = deckIndex - 1;
+    export let notetypeColumn: number = notetypeIndex - 1;
+
+    $: tagsColumn = tagsIndex - 1;
+    $: deckColumn = deckIndex - 1;
+    $: notetypeColumn = notetypeIndex - 1;
 
     $: options = [tr.changeNotetypeNothing()].concat(columnNames);
 </script>
 
-<MapperRow label={tr.editingTags()} {options} bind:index={tagsColumn} />
-<MapperRow label={tr.decksDeck()} {options} bind:index={deckColumn} />
-<MapperRow label={tr.notetypesNotetype()} {options} bind:index={notetypeColumn} />
+<MapperRow label={tr.editingTags()} {options} bind:index={tagsIndex} />
+<MapperRow label={tr.decksDeck()} {options} bind:index={deckIndex} />
+<MapperRow label={tr.notetypesNotetype()} {options} bind:index={notetypeIndex} />
```

Comment:
From your earlier comment:

> Maybe I should remove the selectors for notetype and deck? If there are notetype and deck columns in the CSV, they could still be listed in the dropdown menu, but unselectable.

Yeah, it feels a bit confusing at the moment - having both manual selection in the import options, but also supporting mapping makes it a bit unclear (at least to me) how things will work. Maybe instead of showing these rows, we could just show an info line if deck or notetype have been set in the file? And for the field/tag mapping, are we able to disable certain columns from selection? If not, that might make things tricky if we're relying on array indices.

Reply:
An info sounds good. Should we also disable the deck/notetype selectors if such columns are present? Currently, we use them as fallbacks, if a row leaves these columns empty, but I guess that might be confusing, too. Either all or no notetypes/decks should be allowed to be remapped.
Consequentially, every option that is set in the file should be disabled then. We'd need a `force` flag for every value in the CsvMetadata message, though.

Reply:
Just thought, we probably don't even need info boxes if we go this route. If a widget is greyed out and says "This setting is defined in the chosen file", it's probably clear enough.

As for disabling select options, EnumSelector from deck-options supports this already. But of course, we could also replace the array with an object.

Reply:
+1 on all/none - I don't think the complexity of allowing a fallback is worth it

> We'd need a force flag for every value in the CsvMetadata message, though.

I'm not sure I'm following here - isn't it just deck/notetype we're concerned about?

Reply:
I meant if the file specifies `#delimiter:Colon` or `#is_html:true`, then it doesn't make much sense to let the user change these settings. But currently, `get_csv_metadata()` just returns an enum/boolean, so the frontend code can't know whether it's auto-detected, thus allowed to be changed, or explicitly set in the file, therefore disallowed to be changed by the user.
So the message would need a separate flag `force_is_html` bool to inform about that.

Reply:
Ah, I see what you mean. I don't have strong feelings here - disabling the items that are provided in the file is probably a bit more user friendly, but directives in text files are probably going to be used mainly by power users, who should theoretically understand that it doesn't make sense to change them in the GUI later. The deck/notetype selectors were a bit more of a priority, since we were displaying two different ways to input them, which was somewhat confusing. But no objections to disabling these fields if you think that's best.

Reply:
> but directives in text files are probably going to be used mainly by power users,

I was also planning to write some metadata when exporting. At least `html` and `delimiter` depend solely on the file and not the state of the target collection. If we know these settings for certain we can spare the user the chance to mess things up.